Tutorials/i2c.md

```diff
@@ -0,0 +1,110 @@
+# I2C
+
+I2C is a simple-to-use communication protocol which can have more than one master, with the upper bus speed defined, and only two wires with pull-up resistors are needed to connect almost _unlimited_ number of I2C devices.
+
+Each slave device has a unique address. Transfer from and to master device is serial and it is split into 8-bit packets. All these simple requirements make it very simple to implement I2C interface even with cheap microcontrollers that have no special I2C hardware controller. You only need 2 free I/O pins and few simple I2C routines to send and receive commands.
+
+Let us get the accelerometer values using the I2C protocol. We will be using the [The Tessel Accelerometer](https://www.seeedstudio.com/Tessel-Accelerometer-Module-p-2223.html)
+
+[Refer to the datasheet for the accelerometer module to follow along](http://www.nxp.com/docs/en/data-sheet/MMA8452Q.pdf)
+
+![Fritzing Diagram](http://i.imgur.com/zK4U4S3.png)
+
+```js
+var tessel = require('tessel');
+
+// Connect to device
+var port = tessel.port.A; // Use the SCL/SDA pins of Port A
+
+// This address of the Slave has been taken from https://github.com/tessel/accel-mma84/blob/master/index.js#L15
+// It can also be found in Table 10 - I2C device address sequence on page number 17 of the datasheet. 
+var slaveAddress = 0x1D; // Specific to device
+
+var i2c = new port.I2C(slaveAddress); // Initialize I2C communication
+
+// Details of I2C read and I2C transfer
+var numBytesToRead = 1; // Read back this number of bytes
+
+// Read data over I2C using i2c.read
+i2c.read(numBytesToRead, function (error, dataReceived) {
+  // Print data received (buffer of hex values)
+  console.log('Buffer returned by I2C slave device ('+slaveAddress.toString(16)+'):', dataReceived);
+
+});
+
+// Read/Receive data over I2C using i2c.transfer
+// 0x0D is the WHO_AM_I Register which sends back an acknowledgement to the master for starting the communication
+// Information about 0x0D can also be found in Table 11 - Register address map on page number 19 of the datasheet.
+
+i2c.transfer(new Buffer([0x0D]), numBytesToRead, function (error, dataReceived) {
+    // Print data received (buffer of hex values)
+    // The returned buffer from the I2C slave device should be [0x2A] as specified in the register address map on page number 19 of the MMA84 datasheet
+  console.log('Buffer returned by I2C slave device ('+slaveAddress.toString(16)+'):', dataReceived);
+
+});
+
+/*
```

Comment:
This feels like it should be a break between code examples. The first one tells you "yes, the accelerometer is connected and we have communication"; the second one gets you the actual values. It would be helpful to see the concepts introduced separately

Reply:
(and possibly also helpful for readers to see that that's how it's usually done– you write the simplest possible code first to establish a correct setup, then later write the code that does what you really need)